src/saturn/saturn_projectfile.cpp

```diff
@@ -0,0 +1,450 @@
+    do {
+        data[*offset + i] = string[i];
+        i++;
+    }
+    while (string[i] != 0);
+    *offset += i + 1;
+}
+void put_floating_point(char* data, int* offset, float value) {
+    union {
+        float x;
+        uint y;
+    } u;
+    u.x = value;
+    put_int32(data, offset, u.y);
+}
+void put_bool(char* data, int* offset, bool value) {
+    put_int8(data, offset, value ? 1 : 0);
+}
+void put_identifier(char* data, int* offset, char* identifier) {
+    for (int i = 0; i < SATURN_PROJECT_IDENTIFIER_LENGTH; i++) {
+        data[*offset + i] = identifier[i];
+    }
+    *offset += SATURN_PROJECT_IDENTIFIER_LENGTH;
+}
+void pad(char* data, int* offset, int alignment) {
+    if (*offset % alignment == 0) return;
+    int padding = alignment - *offset % alignment;
+    for (int i = 0; i < padding; i++) {
+        put_int8(data, offset, rand() % 256);
+    }
+}
+
+void align(int* offset, int alignment) {
+    if (*offset % alignment == 0) return;
+    *offset += alignment - *offset % alignment;
+}
+
+uint hash(char* data, int length, int offset = 0) { // See Java's java.lang.String#hashCode() method
+    uint hash = 0;
+    for (int i = 0; i < length; i++) {
+        hash += 31 * hash + (int)(uchar)(data[i + offset]);
+    }
+    return hash;
+}
+
+void saturn_load_project(char* filename) {
+    int pointer = 0;
+    std::ifstream file(filename, std::ios::binary);
+    if (!file.is_open()) return;
+    char* headerData = (char*)malloc(SATURN_PROJECT_HEADER_SIZE);
+    file.read(headerData, SATURN_PROJECT_HEADER_SIZE);
+    char fileIdentifier[5];
+    identifier(headerData, &pointer, fileIdentifier);
+    if (std::strcmp(fileIdentifier, SATURN_PROJECT_IDENTIFIER) != 0) {
+        std::cout << "Identifier doesn't match " SATURN_PROJECT_IDENTIFIER << ". Ignoring." << std::endl;
+    }
+    uint blocks = get_int32(headerData, &pointer);
+    uint checksum = get_int32(headerData, &pointer);
+    uint version = get_int32(headerData, &pointer);
+    char* data = (char*)malloc(SATURN_PROJECT_HEADER_SIZE + blocks * SATURN_PROJECT_BLOCK_SIZE);
+    file.seekg(0);
+    file.read(data, SATURN_PROJECT_HEADER_SIZE + blocks * SATURN_PROJECT_BLOCK_SIZE);
+    uint dataChecksum = hash(data, blocks * SATURN_PROJECT_BLOCK_SIZE, SATURN_PROJECT_HEADER_SIZE);
+    if (checksum != dataChecksum) {
+        std::cout << "Checksum failed: should be " << checksum << " but is " << dataChecksum << ", corruptions or crash might happen!" << std::endl;
+    }
+    k_frame_keys.clear();
+    while (true) {
+        int beginning = pointer;
+        char sectionIdentifier[5];
+        identifier(data, &pointer, sectionIdentifier);
+        uint size = get_int32(data, &pointer);
+        align(&pointer, SATURN_PROJECT_BLOCK_SIZE);
+        if (std::strcmp(sectionIdentifier, SATURN_PROJECT_DONE_IDENTIFIER) == 0) break;
+        if (std::strcmp(sectionIdentifier, SATURN_PROJECT_GAME_IDENTIFIER) == 0) {
+            ushort flags = get_int16(data, &pointer);
```

Comment:
Same issue as lines [100](https://github.com/Llennpie/Saturn/blob/f987112d8bf0845917a5dc8266ab123e64f72400/src/saturn/saturn_projectfile.cpp#L100), [101](https://github.com/Llennpie/Saturn/blob/f987112d8bf0845917a5dc8266ab123e64f72400/src/saturn/saturn_projectfile.cpp#L101), and [138](https://github.com/Llennpie/Saturn/blob/f987112d8bf0845917a5dc8266ab123e64f72400/src/saturn/saturn_projectfile.cpp#L138).